app/serializers/basic_user_serializer.rb

```diff
@@ -23,14 +23,26 @@ def user
     object[:user] || object.try(:user) || object
   end
 
+  # the scope variable passed is based on Guardian.new(current_user), and when
+  # looking at a user profile we could be looking at a different user to the
+  # current one.
+  def user_scope
+    return scope if user_is_current_user
+    @user_scope ||= Guardian.new(user)
+  end
+
+  def user_is_current_user
+    object&.id == scope.user&.id
```

Comment:
`object&.id` seems odd to me because the `object` can never be `nil`

Reply:
Good point, the rest of the user serializer assumes it is there. Maybe some time in the past it wasn't? Will change :)